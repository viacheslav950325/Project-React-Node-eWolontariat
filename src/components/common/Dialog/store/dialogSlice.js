import { createSlice } from '@reduxjs/toolkit';

export const FormType = Object.freeze({
  rejestracja: 'rejestracja',
  zalozonyProfil: 'zalozonyProfil',
  loginDialog: 'loginDialog',
  rejestracjaEmail: 'rejestracjaEmail',
});

const initialState = {
  isOpen: false,
  formType: undefined,
};

const slice = createSlice({
  name: 'dialog',
  initialState,
  reducers: {
    openDialog: (state, action) => {
      return { ...action.payload, isOpen: true };
    },
    closeDialog: (state) => {
      state.isOpen = false;
      state.formType = undefined;
    },
  },
});

export const { openDialog, closeDialog } = slice.actions;

export default slice.reducer;
