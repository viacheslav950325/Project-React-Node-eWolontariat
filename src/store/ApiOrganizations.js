export default class Api {
   
    static getData() {
        const uri = 'https://api.npoint.io/0dd81319d981020bb21f';
        return fetch(uri, {
            method: 'GET'
        });
    }
 }
 