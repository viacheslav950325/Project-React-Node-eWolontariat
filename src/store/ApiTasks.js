export default class Api {
   
    static getData() {
        const uri = 'https://api.npoint.io/8d11637b99831903048a';
        return fetch(uri, {
            method: 'GET'
        });
    }
 }
 