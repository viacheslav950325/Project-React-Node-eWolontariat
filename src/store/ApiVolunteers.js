export default class Api {
   
    static getData() {
        const uri = 'https://api.npoint.io/3f0f98a4cd7eca33c3e9';
        return fetch(uri, {
            method: 'GET'
        });
    }
 }