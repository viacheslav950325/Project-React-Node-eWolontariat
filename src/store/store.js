import { configureStore, getDefaultMiddleware} from "@reduxjs/toolkit";
import taskReducer from "./taskSlice"
import volunteerReducer from "./volunteerSlice";
import dialogReducer from "../components/common/Dialog/store/dialogSlice";
import systemReducer from './systemSlice';
import organizationReducer from './organizationSlice'
//import { composeWithDevTools } from "@reduxjs/toolkit/dist/devtoolsExtension";

//const composedEnhancer = composeWithDevTools(applyMiddleware(thunkMiddleware));

export const store = configureStore({
    middleware: getDefaultMiddleware({
        serializableCheck: false
      }),
    reducer:{
        volunteers: volunteerReducer,
        dialog: dialogReducer,
        system: systemReducer,
        task: taskReducer,
        organization: organizationReducer,
    },
    //composedEnhancer
});

