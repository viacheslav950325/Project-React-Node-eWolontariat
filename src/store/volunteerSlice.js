import { createSlice } from "@reduxjs/toolkit";
import { fetchVolunteers } from "./fetchVolunteers";

const initialState = {
    volunteers: [],
    status: 'idle',
    error: null
  };

const volunteersSlice = createSlice({
    name: 'volunteers',
    initialState,
    reducers: {
        sortVolunteers: (state,action) => {
            state.volunteers.map((volunteer, id)=>{
              let par = volunteer.actions;
              volunteer.numberActions= par.length;
              return volunteer;
              })
            state.volunteers.sort(compare);
            return state;
        },
        addNewVolunteer: (state, action) => {
            state.volunteers.push(action.payload);
        },
        selectVolunteers: (state, action) => {
            return action.payload;
        }
    },
    extraReducers(builder) {
        builder
          .addCase(fetchVolunteers.pending, (state, action) => {
            state.status = 'loading...'
          })
          .addCase(fetchVolunteers.fulfilled, (state, action) => {
            state.status = 'succeeded (:'
            state.volunteers = action.payload;
          })
          .addCase(fetchVolunteers.rejected, (state, action) => {
            state.status = 'failed :('
            state.error = action.error.message
          })
      }
});

export const { sortVolunteers, addNewVolunteer, selectVolunteers, filterVolunteers } = volunteersSlice.actions;

export default volunteersSlice.reducer;

export const selectAllVolunteers = state => state.volunteers;


export const selectVolunteerId = (state, volunteerId) =>
  state.find(volunteer => volunteer.id === volunteerId);
  
function compare( a, b ) {
  if ( a.numberActions > b.numberActions ){
      return -1;
  }
  if ( a.numberActions < b.numberActions ){
      return 1;
  }
    return 0;
}

