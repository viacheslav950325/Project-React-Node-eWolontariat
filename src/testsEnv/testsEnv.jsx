import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import { mount, shallow } from "enzyme";
import * as ReactReduxHooks from "react-redux";

export const testingEnvironment = (defaultStoreState, Component) => {
    const defaultStore = configureStore([thunk])(defaultStoreState);
    const useSelectorSpy = jest.spyOn(ReactReduxHooks, "useSelector");
    const useDispatch = jest.spyOn(ReactReduxHooks, "useDispatch");
  
  
    const shallowWrapper = (store = defaultStore, props) => {
      useSelectorSpy.mockImplementation((state) => state(store.getState()));
      useDispatch.mockImplementation(() => store.dispatch);
  
      return shallow(<Component {...props} />);
    };
  
    const mountWrapper = (store = defaultStore, props) => {
      useDispatch.mockRestore();
  
      return mount(
        <Provider store={store}>
          <Component {...props} />
        </Provider>
      );
    };
    return { defaultStore, shallowWrapper, mountWrapper };
  };