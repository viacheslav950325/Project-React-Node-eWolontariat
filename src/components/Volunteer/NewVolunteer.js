import { Box,Button,TextField } from "@material-ui/core"
import { useState } from "react";
import { useDispatch } from "react-redux";
import { addNewVolunteers } from "../../store/volunteerSlice";

const NewVolunteer = () => {
    const [value, setValue] = useState("");
    const dispatch = useDispatch();
    return(
        <Box>
            <TextField 
                onChange={(e)=>setValue(e.target.value)}
                variant="outlined"
                value={value}>
            </TextField>
            <Button disabled={value.trim() === ""}
                 onClick = {() => {
                    dispatch(addNewVolunteers(value));
                    setValue("");
                 }

                 }>Add
            </Button>   
        </Box>
    );
};

export default NewVolunteer;