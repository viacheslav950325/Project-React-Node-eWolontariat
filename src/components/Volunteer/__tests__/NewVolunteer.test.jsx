import { testingEnvironment } from "../../../testsEnv/testsEnv";
import NewVolunteer from "../NewVolunteer";
import { Button, TextField } from "@material-ui/core";
import { act } from "@testing-library/react";
import { addNewVolunteer } from "../../../store/volunteerSlice";

const { shallowWrapper, mountWrapper, defaultStore } = testingEnvironment(
  {},
  NewVolunteer
);

describe("NewVolunteer", () => {
  beforeEach(() => {
    defaultStore.clearActions();
  });
  it("Should render correctly", () => {
    const wrapper = shallowWrapper();
    expect(wrapper).toMatchSnapshot();
  });
  it("Should dispatch action on add new volunteer", () => {
    const wrapper = mountWrapper();
    act(() => {
      wrapper
        .find(TextField)
        .props()
        .onChange({ target: { value: "test" } });
    });
    wrapper.update();
    wrapper.find(Button).simulate("click");
    expect(defaultStore.getActions()[0].type).toEqual(addNewVolunteer.type);
  });
});
