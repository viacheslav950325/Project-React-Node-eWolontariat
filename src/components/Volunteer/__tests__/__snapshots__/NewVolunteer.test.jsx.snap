// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewVolunteer Should render correctly 1`] = `
<Styled(MuiBox)>
  <WithStyles(ForwardRef(TextField))
    onChange={[Function]}
    value=""
    variant="outlined"
  />
  <WithStyles(ForwardRef(Button))
    disabled={true}
    onClick={[Function]}
  >
    Add
  </WithStyles(ForwardRef(Button))>
</Styled(MuiBox)>
`;
